File: Practica1/ej4/Estadistica/estadistica.py
import math
def promedio(datos):
    suma=0
    for i in range(len(datos)):
        suma+=datos[i]
    prom=suma/len(datos)
    return prom
def desviacionestandar(datos):
    suma=0
    prom=promedio(datos)
    for i in range(len(datos)):
        suma+=(datos[i]-prom)**2
    desv = math.sqrt (suma/(len(datos)-1))
    return desv

datos=[0]*10
for i in range (len(datos)):
    datos[i]=float(input())

print("El promedio es ",promedio(datos))
print("La desviacion estandar es ", desviacionestandar(datos))

File: Practica1/ej4/Estadistica/src/Estadistica.java
import java.util.Scanner;

public class Estadistica {
    private float[] datos;

    public Estadistica(float[] datos) {
        this.datos = datos;
    }

    public float promedio() {
        float suma = 0;
        for (float dato : datos) {
            suma += dato;
        }
        return suma / datos.length;
    }

    public float desviacionEstandar() {
        float prom = promedio();
        float sumaDiferenciasCuadrado = 0;

        for (float dato : datos) {
            sumaDiferenciasCuadrado += Math.pow(dato - prom, 2);
        }

        return (float) Math.sqrt(sumaDiferenciasCuadrado / (datos.length - 1));
    }

    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        float[] numeros = new float[10];

        System.out.println("Ingresa 10 números:");
        for (int i = 0; i < 10; i++) {
            numeros[i] = entrada.nextFloat();
        }

        Estadistica estadistica = new Estadistica(numeros);

        System.out.println("El promedio es: " + estadistica.promedio());
        System.out.println("La desviación estándar es: " + estadistica.desviacionEstandar());

        entrada.close();
    }
}